### PR TITLE
DENG-618, DENG-619. Collect all first_session pings and more attribution from metrics ping

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
DENG-618, DENG-619. Collect all first_session pings and more attribution from metrics ping.

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [x] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql [hunk 1]
```diff
@@ -200,6 +207,10 @@ SELECT
       _previous.metadata.min_first_session_ping_run_date,
       _current.metadata.min_first_session_ping_run_date
     ) AS min_first_session_ping_run_date,
+    COALESCE(
+      _previous.metadata.min_metrics_ping_submission_date,
+      _current.metadata.min_metrics_ping_submission_date
+    ) AS min_metrics_ping_submission_date,
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:213]: This should take the min:
```suggestion
    LEAST(
      _previous.metadata.min_metrics_ping_submission_date,
      _current.metadata.min_metrics_ping_submission_date
    ) AS min_metrics_ping_submission_date,
```

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql [hunk 2]
```diff
@@ -54,9 +51,18 @@ metrics_ping AS (
...
+    ] AS adjust_campaign,
     ARRAY_AGG(metrics.string.metrics_adjust_network ORDER BY submission_timestamp ASC)[
       SAFE_OFFSET(0)
     ] AS adjust_network,
+    ARRAY_AGG(metrics.string.metrics_adjust_ad_group ORDER BY submission_timestamp ASC)[
+      SAFE_OFFSET(0)
+    ] AS adjust_ad_group,
+    ARRAY_AGG(metrics.string.metrics_adjust_creative ORDER BY submission_timestamp ASC)[
+      SAFE_OFFSET(0)
+    ] AS adjust_creative,
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:65]: These need to `IGNORE NULLS`

**Review by fbertsch — CHANGES_REQUESTED** [review]

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql (hunk 1 above)
Comment (lucia-vargas-a) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:213]: The problem with LEAST is that it returns null as minimun if any value is null. We'd need a CASE,

Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:213]: Yeah, then we'll have to handle nulls. But we definitely want the minimum

**Review by fbertsch — APPROVED** [review]
LGTM!

**Pull request merged**